component/answer/src/main/java/org/exoplatform/management/answer/operations/AnswerDataImportResource.java

```diff
@@ -288,5 +299,40 @@ private void removeCategoryQuestions(Category category, List<Question> questions
       log.error("Fail to remove questions of category: "+category.getName(),e);
     }
   }
+  private void removeNewCategories(List<Object> objectList,Category category){
+    try {
+      Category toReplaceCategory = faqService.getCategoryById(category.getId());
+      if(toReplaceCategory==null){
+        return;
+      }
+      List<Category> targetCategoriesList = faqService.getSubCategories(category.getPath(),AnswerExtension.EMPTY_FAQ_SETTIGNS,true,null);
+      if(targetCategoriesList==null || targetCategoriesList.size()==0){
+        return;
+      }
+      List<Category> sourceCategoriesList = new ArrayList<Category>();
+      String categoryPath = category.getPath();
+      for(int i=0; i<objectList.size(); i+=2){
+        Category sub_category = (Category) objectList.get(i);
+        if(getParentPath(sub_category).equals(categoryPath)){
+          sourceCategoriesList.add(sub_category);
+        }
+      }
+      for(Category sub_category : targetCategoriesList){
+        if(!sourceCategoriesList.contains(sub_category)){
+          log.info("Delete the category: '" + sub_category.getName() + "' because it's only founded here (delete-newcategories=true)");
+          deleteActivities(sub_category.getId(), null);
+          faqService.removeCategory(sub_category.getPath());
+          if (faqService.getCategoryById(sub_category.getId()) != null) {
```

Comment:
No need to check and throw an exception if the category is not removed, if there is a problem it will be catched in the catch block below.